src/main/java/mate/academy/springboot/swagger/controller/ProductController.java

```diff
@@ -0,0 +1,86 @@
+package mate.academy.springboot.swagger.controller;
+
+import java.math.BigDecimal;
+import java.util.List;
+import mate.academy.springboot.swagger.dto.ProductDto;
+import mate.academy.springboot.swagger.mapper.Mapper;
+import mate.academy.springboot.swagger.model.Product;
+import mate.academy.springboot.swagger.service.ProductService;
+import mate.academy.springboot.swagger.service.SortParamParser;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Sort;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/products")
+public class ProductController {
+    private final ProductService productService;
+    private final SortParamParser sortParamParser;
+    private final Mapper mapper;
+
+    public ProductController(ProductService productService, SortParamParser sortParamParser,
+                             Mapper mapper) {
+        this.productService = productService;
+        this.sortParamParser = sortParamParser;
+        this.mapper = mapper;
+    }
+
+    @PostMapping
+    public ResponseEntity<ProductDto> add(@RequestBody ProductDto productDto) {
```

Comment:
Add annotation `@ApiOperation` and document every public method of your API

Reply:
done